bindings/java/src/main/java/org/cssinline/CssInlineConfig.java

```diff
@@ -0,0 +1,65 @@
+package org.cssinline;
+
+public class CssInlineConfig {
+  public final boolean inlineStyleTags;
+  public final boolean keepStyleTags;
+  public final boolean keepLinkTags;
+  public final boolean loadRemoteStylesheets;
+  public final String  baseUrl;
+  public final String  extraCss;
+  public final int     cacheSize;
+  public final int     preallocateNodeCapacity;
+
+  private CssInlineConfig(
+    boolean inlineStyleTags,
+    boolean keepStyleTags,
+    boolean keepLinkTags,
+    boolean loadRemoteStylesheets,
+    String baseUrl,
+    String extraCss,
+    int cacheSize,
+    int preallocateNodeCapacity
+  ) {
+    this.inlineStyleTags = inlineStyleTags;
+    this.keepStyleTags = keepStyleTags;
+    this.keepLinkTags = keepLinkTags;
+    this.loadRemoteStylesheets = loadRemoteStylesheets;
+    this.baseUrl = baseUrl;
+    this.extraCss = extraCss;
+    this.cacheSize = cacheSize;
+    this.preallocateNodeCapacity = preallocateNodeCapacity;
+  }
+
+  public static class Builder {
+    private boolean inlineStyleTags = true;
+    private boolean keepStyleTags   = false;
+    private boolean keepLinkTags    = false;
+    private boolean loadRemoteStylesheets = true;
+    private String  baseUrl         = null;
+    private String  extraCss        = null;
+    private int     cacheSize       = 0;
+    private int     preallocateNodeCapacity = 32;
+
+    public Builder setInlineStyleTags(boolean b)        { this.inlineStyleTags = b; return this; }
+    public Builder setKeepStyleTags(boolean b)          { this.keepStyleTags   = b; return this; }
+    public Builder setKeepLinkTags(boolean b)           { this.keepLinkTags    = b; return this; }
+    public Builder setLoadRemoteStylesheets(boolean b)  { this.loadRemoteStylesheets = b; return this; }
+    public Builder setBaseUrl(String url)               { this.baseUrl = url;   return this; }
```

Comment:
It would be nice to validate `Url` here to, so it would be impossible to create an invalid config instance.